### PR TITLE
Validator testsuite

### PR DESCRIPTION
**Pull request by AshakaE**
# Validator service test suite

_Test suite allows testing of random JSON of Wiki type to ensure validator service works as expected_

## How should this be tested?

1. `yarn test --verbose validator.service.spec.ts`

## Linked issues

closes [#344](https://github.com/EveripediaNetwork/issues/issues/344)

### CODE REVIEW

File: src/App/pinJSONAndImage/pin.service.ts [hunk 1]
```diff
@@ -40,7 +41,7 @@ class PinService {
   }
 
   async pinJSON(body: string): Promise<IpfsHash | any> {
-    const data = JSON.parse(`${body}`)
+    const data: ValidWiki = JSON.parse(`${body}`)
```
Comment (kesar) [inline src/App/pinJSONAndImage/pin.service.ts:44]: is there any reason to cast a string to string? 😕 

```suggestion
    const data: ValidWiki = JSON.parse(body)
```

File: src/Indexer/Validator/validator.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import { Test, TestingModule } from '@nestjs/testing'
```
Comment (kesar) [inline src/Indexer/Validator/validator.service.spec.ts:1]: awesome test! <3

**Review by kesar — APPROVED** [review]
only a small code suggestion but good to go.

make sure pipelines execute this test

File: src/App/pinJSONAndImage/pin.service.ts (hunk 1 above)
Comment (kesar) [reply src/App/pinJSONAndImage/pin.service.ts:44]: if its not a string typescript wont let him pass a function param in : pinJSON(body: string)

so its checking the type string in the function param, then doing some weird cast to string using literals. Not big deal but I am fairly confident that can be simplified w the suggestion. Let's not stop this merge but take a look if you want later 👍🏻

Comment (AshakaE) [reply src/App/pinJSONAndImage/pin.service.ts]: oh okay, I see what you mean ...I'll look at it later 🤔

**Pull request merged**